app/domain/authentication/authn_jwt/identity_providers/identity_from_decoded_token_provider.rb

```diff
@@ -0,0 +1,65 @@
+module Authentication
+  module AuthnJwt
+    module IdentityProviders
+      # Class for providing jwt identity from the decoded token from the field specified in a secret
+      class IdentityFromDecodedTokenProvider < IdentityProviderInterface
+        def initialize(authentication_parameters)
+          @logger = Rails.logger
+
+          @secret_fetcher = Conjur::FetchRequiredSecrets.new
+          @resource_class = ::Resource
+          @authentication_parameters = authentication_parameters
+          @resource_id = @authentication_parameters.authn_jwt_variable_id
+          @decoded_token = @authentication_parameters.decoded_token
+        end
+
+        def jwt_identity
+          return @jwt_identity if @jwt_identity
+
+          token_field_name = fetch_token_field_name
+          @logger.debug(LogMessages::Authentication::AuthnJwt::CheckingIdentityFieldExists.new(token_field_name))
+          @jwt_identity ||= @decoded_token[token_field_name]
+          if @jwt_identity.blank?
+            raise Errors::Authentication::AuthnJwt::NoSuchFieldInToken, token_field_name
+          end
+
+          @logger.debug(LogMessages::Authentication::AuthnJwt::FoundJwtFieldInToken.new(token_field_name, jwt_identity))
```

Comment:
this will call the memoizing method again instead of writing the object itself
```suggestion
          @logger.debug(LogMessages::Authentication::AuthnJwt::FoundJwtFieldInToken.new(token_field_name, @jwt_identity))
```